bokehjs/src/coffee/models/canvas/canvas.coffee

```diff
@@ -47,15 +47,23 @@ class CanvasView extends BokehView
     ctx = canvas_el[0].getContext('2d')
     return ctx
 
+  need_render: () ->
+    # Public method so the plot can determine if size/dpi changed
+    width = @mget('width')
+    height = @mget('height')
+    dpr = window.devicePixelRatio  # this changes when browser zoom changes
+    return not _.isEqual(@last_dims, [width, height, dpr])
```

Comment:
I'm a bit confused, I thought `canvas.set_dims` does not update `model.width` and `model.height` - facts to the contrary gratefully appreciated. If that's true, doesn't that mean that any time an object is responsive `model.width` and `model.height` will have gotten out of sync with `@last_dims` - even under current responsive code?